File: src/triallife/shared/interfaces/vehicle.ts
import { Vector3 } from './vector';

export interface Vehicle {
    uid: string;
    model: string;
    position: Vector3;
    rotation: Vector3;
    fuel?: number;
    color?: {
        r: number;
        g: number;
        b: number;
        a: number;
    };
}

File: src/core/server/Database/Models/Vehicle.cs
using AltV.Net.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;
using System.Text;
using System.Threading.Tasks;

namespace triallife.Database.Models {
	public class Vehicle {
		public int id { get; set; }
		public string model { get; set; }
		public Vector3 position { get; set; }
		public Vector3 rotation { get; set; }
		public double fuel { get; set; }
		public Rgba color { get; set; }
		//mods etc
	}
}
